File: src/test/baekjoon/python/greedy/Assignment.py
import heapq

N = int(input())
arr = []
for _ in range(N):
    day, score = map(int, input().split())
    arr.append([-score, day, score])

heapq.heapify(arr)

answer = [0] * 1001

while arr:
    items = heapq.heappop(arr)
    for i in range(items[1], 0, -1):
        if answer[i] == 0:
            answer[i] = items[2]
            break

print(sum(answer))

File: src/test/baekjoon/java/greedy/Assignment.java
package test.baekjoon.java.greedy;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.PriorityQueue;
import java.util.StringTokenizer;

public class Assignment {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;
        int N = Integer.parseInt(br.readLine());
        PriorityQueue<int[]> arr = new PriorityQueue<>((o1, o2) -> o2[1] - o1[1]);

        for(int i = 0; i < N; i++) {
            st = new StringTokenizer(br.readLine());
            int day = Integer.parseInt(st.nextToken());
            int score = Integer.parseInt(st.nextToken());
            arr.add(new int[]{day, score});
        }


        int[] answer = new int[1001];
        while(!arr.isEmpty()) {
            int[] items = arr.poll();
            for(int i = items[0]; i > 0; i--) {
                if(answer[i] == 0) {
                    answer[i]= items[1];
                    break;
                }
            }
        }

        System.out.println(Arrays.stream(answer).sum());
    }
}